tests/BridgeImplementationTest.php

```diff
@@ -1,188 +1,204 @@
 <?php
-
-use PHPUnit\Framework\TestCase;
-use PHPUnit\Framework\TestResult;
-use PHPUnit\Framework\AssertionFailedError;
-
 require_once __DIR__ . '/../lib/rssbridge.php';
 
-/**
- * This class checks bridges for implementation details:
- *
- * - A bridge must not implement public functions other than the ones specified
- *   by the bridge interfaces. Custom functions must be defined in private or
- *   protected scope.
- * - getName() must return a valid string (non-empty)
- * - getURI() must return a valid URI
- * - A bridge must define constants for NAME, URI, DESCRIPTION and MAINTAINER,
- *   CACHE_TIMEOUT and PARAMETERS are optional
- */
-final class BridgeImplementationTest extends TestCase {
-	private function CheckBridgePublicFunctions($bridgeName){
-
-		$parent_methods = array();
-
-		if(in_array('BridgeInterface', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('BridgeInterface'));
-		}
-
-		if(in_array('BridgeAbstract', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('BridgeAbstract'));
-		}
-
-		if(in_array('FeedExpander', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('FeedExpander'));
-		}
-
-		// Receive all non abstract methods
-		$methods = array_diff(get_class_methods($bridgeName), $parent_methods);
-		$method_names = implode(', ', $methods);
+use PHPUnit\Framework\TestCase;
 
-		$errmsg = $bridgeName
-		. ' implements additional public method(s): '
-		. $method_names
-		. '! Custom functions must be defined in private or protected scope!';
+class BridgeImplementationTest extends TestCase {
+	const PATH_BRIDGES = __DIR__ . '/../bridges/';
 
-		$this->assertEmpty($method_names, $errmsg);
+	private $class;
+	private $obj;
 
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testClassName($path) {
+		$this->setBridge($path);
+		$this->assertTrue($this->class === ucfirst($this->class), 'class name must start with uppercase character');
 	}
```

Comment:
Bridge names must not contain spaces and must end with "Bridge"
See also: https://github.com/RSS-Bridge/rss-bridge/wiki/How-to-create-a-new-Bridge

```suggestion
		$this->assertEquals(0, substr_count($this->class, ' '), 'class name must not contain spaces');
		$this->assertStringEndsWith('Bridge', $this->class, 'class name must end with "Bridge"');
	}
```